src/compiler/ir/README.md

```diff
@@ -101,7 +101,122 @@ This could be refined by considering, which classes could reach `i`.  For instan
 
 # Region Tracking
 
-[TODO]
+We need the provenance of each mutable reference to an object to its 
+region to be represented in the code.  
+This is only required for variable that can contain a mutable reference into region tree.
+We need this because we support certain operations on a region that can be done by having
+a reference to any particular object in that region.
+
+An `iso` reference is both a handle on an object, and can cheaply access its region
+information.
+
+A `mut` reference does not cheaply know its owning region, 
+so we need the compiler to have a second
+reference associate with each `mut` that refers to its region.
+
+To maintain this data, for an aliasing field read
+```
+y = x.f
+``` 
+we need translate this to
+```
+y = // Read field described below.
+if (y->rt::Object::get_class() == RegionMD::ISO)
+  y_region = y
+else 
+  y_region = x_region
```

Comment:
This needs x_region to not be null as well, just like the allocation stuff further down.

Why did you go with the region is null, rather than tagging the low bits as you’d mention previously?

Reply:
I like the idea of using the front-end to separate out allocation on an `iso` reference from `mut`, and then this seemed cleaner.

We can put the alternative in.